analyzer/src/traversal/functions.rs

```diff
@@ -355,10 +355,10 @@ fn emit(
     } = &stmt.kind
     {
         return match kind {
-            fe::Expr::Call { func, args } => {
-                let event_name = &expressions::expr_name_string(func)?;
+            fe::Expr::Call { func, args } if matches!(&func.kind, fe::Expr::Name(_)) => {
```

Comment:
This feels like a band aid fix and builds more evidence that we should change the representation of `emit` in our AST. It's currently represented as:

```
Emit {
        value: Node<Expr>,
    },
```

But at least for now the right hand side can not be any `Expr` so it should probably not be represented as such but rather as something with just `name` and `args`.

Unless..at some point we would allow something like:

```
pub def some_action():
  emit get_actual_event()

def get_actual_event(some_condition) -> MyEvent:
  # Some checks
  return MyEvent(..)
```

But it's not clear to me if that would ever be possible (e.g. it isn't in solidity afaik).

Anyway, I did not go for the AST and parser change but for this band aid instead because its probably not worth investing time into parser changes when its going to be rewritten anyway.

I will open an issue about the parser change instead.

Reply:
Created an issue here https://github.com/ethereum/fe/issues/337